test/test-network/systemd-networkd-tests.py

```diff
@@ -2176,6 +2177,42 @@ def test_dhcp_client_gateway_onlink_implicit(self):
         print(output)
         self.assertRegex(output, 'onlink')
 
+    def test_dhcp_client_with_ipv4ll_fallback(self):
+        self.copy_unit_to_networkd_unit_path('25-veth.netdev', 'dhcp-server-veth-peer.network',
+                                             'dhcp-client-with-ipv4ll-fallback.network')
+        self.start_networkd(0)
+        self.start_dnsmasq(lease_time='2m')
```

Comment:
I am trying to figure out the test . Do we need to run the dnsmasq at all ? I think we need to wait for ```MAX_CLIENT_ATTEMPT``` expires then we need to verify ```169.254``` address is assigned to the interface /

Reply:
The test is for the case that DHCP works fine. In that case, IPv4LL address should not be assigned, and if the test pass, the address is actually not assigned.

I will try to add a test case that without DHCP server (dnsmasq), hence the case IPv4LL address will be assigned. But I'd like to do that in other later PR.